Clarify functionality comments:

jacobweinstock opened the pull request:
## Description

<!--- Please describe what this PR is going to change -->
This PR addresses feedback given in #2 after its merge. This tries to help clarify some things around option 43 suboptions and around the use of non-string type arguments for bootfileAndNextServer.

## Why is this needed

<!--- Link to issue you have raised -->

Fixes: #

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->


## How are existing users impacted? What migration steps/scripts do we need?

<!--- Fixes a bug, unblocks installation, removes a component of the stack etc -->
<!--- Requires a DB migration script, etc. -->


## Checklist:

I have:

- [ ] updated the documentation and/or roadmap (if required)
- [ ] added unit or e2e tests
- [ ] provided instructions on how to upgrade

jacobweinstock committed:
Clarify functionality comments:

This tries to help clarify some things around
option 43 suboptions and around the use of non-string
type arguments for bootfileAndNextServer.

Signed-off-by: Jacob Weinstock <jakobweinstock@gmail.com>

nshalman commented on option.go:
```diff
@@ -116,12 +118,12 @@ func (s *Server) setDHCPOpts(_ context.Context, _ *dhcpv4.DHCPv4, d *data.Dhcp)
 // option 60: Class Identifier. https://www.rfc-editor.org/rfc/rfc2132.html#section-9.13
 // option 60 is set if the client's option 60 (Class Identifier) starts with HTTPClient.
-func (s *Server) setNetworkBootOpts(ctx context.Context, m *dhcpv4.DHCPv4, n *data.Netboot) func(d *dhcpv4.DHCPv4) {
+func (s *Server) setNetworkBootOpts(ctx context.Context, m *dhcpv4.DHCPv4, n *data.Netboot) dhcpv4.Modifier {
```
oooh, Very nice! :)

nshalman commented on option.go:
```diff
@@ -157,6 +159,8 @@ func (s *Server) setNetworkBootOpts(ctx context.Context, m *dhcpv4.DHCPv4, n *da
 
 // bootfileAndNextServer returns the bootfile (string) and next server (net.IP).
+// input arguments `tftp`, `ipxe` and `iscript` use non string types so as to attempt to be more clear about the expectation around what is wanted for these values.
+// It also helps us avoid having to validate a string in multiple ways.
```
These are helpful, but I also curious about the formatting being done to generate things like `bootfile` and `bin`.
I should have been more clear.

jacobweinstock replied:
would you mind elaborating on this? The curiosity is not around the return types but the argument types in relation to how they are used to populate the return values?

nshalman replied:
Per our verbal conversation, this is around how we're wiring up "system level" observability by encoding interesting information in our responses. I think you'll address that later, so perhaps we can just merge this PR as-is.

nshalman reviewed (dismissed):
I still have questions, but this is all net-win. :)

jacobweinstock committed:
Remove mac address from bootfile name:

This was an attempt to do some correlation between
DHCP and tftp/http ipxe servers but was more confusing
than its worth. Also, using otel going forward seems like
a better option.

Signed-off-by: Jacob Weinstock <jakobweinstock@gmail.com>
